### PR TITLE
fix: External table models with missing fields now produce an error

### PR DESCRIPTION
**Pull request by ericf-firebolt**
### Description

Previously, external table models with missing or misspelled `regex` or `data_type` fields caused indecipherable errors. Those errors are now clear. In addition, drop/create on external tables only occurs if variable is set: `dbt run-operation stage_external_sources  --vars "ext_full_refresh: true"`. Otherwise, table is skipped.

### Fixes
[dbt seeds doesn't truncate tables before running](https://packboard.atlassian.net/browse/FIR-10985)

### Checklist

- [X] I have run this code in development and it appears to resolve the stated issue.
- [X] This PR includes tests, or tests are not required/relevant for this PR.
- [X] I have updated `CHANGELOG.md` and added information about my change.
- [X] If this PR requires a new PyPI release I have bumped the version number.
- [X] I have pulled/merged from the main branch if there are merge conflicts.
- [X] I have verified that this PR contains only code changes relevant to this PR.

### CODE REVIEW

**Comment by octavianzarzu30** [comment]
hey @ericf-firebolt 
1. when running 
```% dbt run-operation stage_external_sources   ```
I get:
```
10:25:29  Running with dbt=1.0.1
10:25:29  Unable to do partial parsing because config vars, config profile, or config target have changed
10:25:30  12:25:30 + 1 of 1 START external source s3.raw_customers
10:25:34  12:25:34 + 1 of 1 (1) DROP TABLE IF EXISTS raw_customers
10:25:34  12:25:34 + 1 of 1 (1) OK
10:25:34  12:25:34 + 1 of 1 (2) --    CREATE EXTERNAL TABLE raw_customers ("id" in...  
10:25:35  12:25:35 + 1 of 1 (2) OK
```

Shouldn't it skip in this case? 

2. Using `false` for the flag, I get a runtime error:
```% dbt run-operation stage_external_sources   --vars "ext_full_refresh: false" ```

```
10:25:45  Running with dbt=1.0.1
10:25:45  Unable to do partial parsing because config vars, config profile, or config target have changed
10:25:46  12:25:46 + 1 of 1 START external source s3.raw_customers
10:25:50  12:25:50 + 1 of 1 (1) --    CREATE EXTERNAL TABLE raw_customers ("id" in...  
10:25:50  Encountered an error while running operation: Runtime Error
  Runtime Error
    Error executing query:
    View/Table 'raw_customers' already exists.
```

3. Using `true` for the flag returns correctly (table is first dropped):
```% dbt run-operation stage_external_sources   --vars "ext_full_refresh: false" ```

```
10:29:33  Running with dbt=1.0.1
10:29:33  Unable to do partial parsing because config vars, config profile, or config target have changed
10:29:34  12:29:34 + 1 of 1 START external source s3.raw_customers
10:29:38  12:29:38 + 1 of 1 (1) DROP TABLE IF EXISTS raw_customers
10:29:38  12:29:38 + 1 of 1 (1) OK
10:29:38  12:29:38 + 1 of 1 (2) --    CREATE EXTERNAL TABLE raw_customers ("id" in...  
10:29:39  12:29:39 + 1 of 1 (2) OK
```

4. I'm not quite sure why the `Unable to do partial parsing because config vars, config profile, or config target have changed` warning keeps popping up

**Comment by octavianzarzu30** [comment]
Yep - pip3 install git+https://github.com/firebolt-db/dbt-firebolt.git@external-table-edits 
and ran - the table existed before. Let's check tomorrow. If it's not reproducible on your end, then I'm good to go. @todd-firebolt can you give it a try as well?

File: dbt/adapters/firebolt/impl.py [hunk 1]
```diff
@@ -159,19 +159,41 @@ def make_field_partition_pairs(self, columns, partitions) -> List[str]:
         unpartitioned_columns = []
         partitioned_columns = []
         for column in columns:
-            unpartitioned_columns.append(
-                self.quote(column['name']) + ' ' + column['data_type']
-            )
+            if 'data_type' in column and column['data_type'] is not None:
```
Comment (stepansergeevitch) [inline dbt/adapters/firebolt/impl.py:162]: nit:
```
if column.get("data_type") is None:
    raise ...
# do stuff    
```

File: dbt/adapters/firebolt/impl.py [hunk 2]
```diff
@@ -159,19 +159,41 @@ def make_field_partition_pairs(self, columns, partitions) -> List[str]:
...
-                partitioned_columns.append(
-                    self.quote(partition['name'])
-                    + ' '
-                    + partition['data_type']
-                    + " PARTITION ('"
-                    + partition['regex']
-                    + "')"
-                )
+                # Todo: See if name is a required field.
+                if 'name' in partition and partition['name'] is not None:
```
Comment (stepansergeevitch) [inline dbt/adapters/firebolt/impl.py:173]: Same nit here. Validate first and then do what is needed

**Review by stepansergeevitch — APPROVED** [review]
lgtm

**Review by stepansergeevitch — COMMENTED** [review]
I get an internal error
```
(.venv) ~/code/jaffle_shop_firebolt [main|…1]$ dbt run-operation stage_external_sources
12:11:04  Running with dbt=1.0.3
12:11:04  1 of 1 START external source s3.raw_customers
12:11:09  1 of 1 (1) DROP TABLE IF EXISTS raw_customers
12:13:20  Encountered an error while running operation: Runtime Error
  Runtime Error
    Error executing query:
    Query failed - Internal execution error
```

File: dbt/adapters/firebolt/impl.py (hunk 1 above)
Comment (ericf-firebolt) [reply dbt/adapters/firebolt/impl.py]: Oh, you're right; that's much better.

**Pull request merged**